src/i2c.rs

```diff
@@ -88,12 +88,12 @@ macro_rules! busy_wait {
         loop {
             let isr = $i2c.isr.read();
 
-            if isr.berr().is_error() {
```

Comment:
I'm unsure about this change though. This change assumes that the variant flag is only set, when no error occured, otherwise errors would be ignored. 

I do not if this is the case, though. I'm currently reading into the documentation, but maybe you can point me into the right direction :)

Reply:
My understanding of the purpose of this macro is to first check whether the variant criterion is satisfied, and secondarily, if the criterion is not satisfied, check for errors.  My change to the macro crystallizes these priorities.

Reply:
I'm not too attached to this change-- just let me know if you want me to revert it.